### PR TITLE
#167891576 As a user, I can log into the Barefoot Nomad application via email & password

### PR DESCRIPTION
**Pull request by Pomile**
#### What does this PR do?
Have a user login
#### Description of Task to be completed?
- Write unit test for login
- Update api doc
- Create login validator
- Create login service
- Create login route handler
- Modify travis
- Install bcrypt
#### How should this be manually tested?
1. clone the project
2. cd to the project directory
3. Run `npm install`
4. Create database in postgres
5. Run `npm run migrate`
6. Run `npm run seed`
7. Run `npm test`

#### What are the relevant pivotal tracker stories?
[#167891576](https://www.pivotaltracker.com/story/show/167891576)
#### Any background context you want to add?
Nil
#### Screenshots
Nil

### CODE REVIEW

File: src/controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import { Users } from '../database/models';
+
+dotenv.config();
+const SECRET = process.env.TOKEN_SECRET;
+
+class Auth {
+  static async login(req, res) {
```
[the remark this answers is not in the document]
Comment (femitj) [reply src/controllers/auth.js]: change the file-name to Auth.js

**Review by femitj — COMMENTED** [review]
can you move your login unit test into a folder called auth.test.js

File: src/controllers/auth.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const { email, password, companyId } = req.body;
```
Comment (thatkazuk1) [inline src/controllers/auth.js]: Do we need `companyId` for login also?

Comment (Pomile) [reply src/controllers/auth.js]: Yes

File: src/controllers/Auth.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const { email, password, companyId } = req.body;
+
+    const user = await User.findOne({ where: { email, companyId } });
+    if (!user) {
+      res.status(404).json({ status: 404, msg: 'incorrect email or password or company code', success: false }).end();
+    } else {
+      const hash = user.password;
+      const result = await bcrypt.compareSync(password, hash);
+      if (result === true) {
+        const payload = user;
```
Comment (tvpeter) [inline src/controllers/Auth.js]: I kind of think the payload is too big, we can select just properties like userid, name and companyId to save in the token

**Review by tvpeter — COMMENTED** [review]
Nice work. Kindly check my below comment.

**Review by tvpeter — COMMENTED** [review]
I just saw the `Token` helper class that you can even use.

**Comment by Pomile** [comment]
> I just saw the `Token` helper class that you can even use.

Thank you Peter,
I am aware that Tijani created Jwt helper which i will latter use when i rebase to the latest branch.

File: src/controllers/Auth.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import { User } from '../database/models';
+
+dotenv.config();
+
+class Auth {
+  static async login(req, res) {
+    const { email, password, companyId } = req.body;
```
Comment (sneakymaxy) [inline src/controllers/Auth.js]: Please could you clarify why ```companyId``` is necessary for login

File: test/auth.test.js [hunk 5]
```diff
@@ -223,4 +222,90 @@ describe('Auth Routes', () => {
         });
     });
   });
+    describe('User Login', () => {
```
[the remark this answers is not in the document]
Comment (thatkazuk1) [reply test/auth.test.js]: Check this indentation issue

**Review by thatkazuk1 — APPROVED** [review]

File: src/database/models/user.js [hunk 6]
```diff
@@ -16,11 +16,7 @@ module.exports = (sequelize, DataTypes) => {
   );
   User.associate = (models) => {
     // associations can be defined here
-    User.belongsTo(models.Company, {
```
Comment (max-wel) [inline src/database/models/user.js:19]: Why did you delete this

**Review by max-wel — CHANGES_REQUESTED** [review]

**Review by max-wel — APPROVED** [review]

**Review by daniellamarr — APPROVED** [review]
Good to go!

**Pull request merged**